### DISCUSSION TITLE
Consolidate addons and settings

### DISCUSSION THREAD

**Discussion by Secret-chest** [ScratchAddons/ScratchAddons#6007] [Discussion]
Here are my suggestions:

## Merge some addons
* Merge `mute-project` and `vol-slider` into `vol-slider`:
    * They are very related, both control the volume - they even use the same icons
    * There would be an option to have the slider
* Merge `custom-block-text`, `custom-block-shape`, `editor-theme3`, `zebra-striping`, `transparent-orphans` into `custom-block-appearance`:
    * All are block themes
    * All can be setup to have no effect already, so there's no need for any toggles
* Merge `block-cherry-picking` and `block-duplicate` into `more-block-controls`:
    * These addons work best together
    * We can have toggles for each, or not
    * All add more controls to block moving
* Merge `mouse-pos`, `clones` and `debugger` into `debugger`:
    * They are used for the same thing
    * There would be toggles for each feature
* Merge `editor-unshare-button` and `share-through-mystuff` into `share-unshare`:
    * They both make share/unshare buttons consistent
    * Should have toggles
* Merge `youtube-fullscreen` and `true-youtube-links` into `youtube`:
    * There's no point in having the former enabled with the latter
    * There would be toggles
* Merge `confirm-actions` and `remove-sprite-confirm` into `confirm-actions`:
    * They're for different areas, but from a useability perspective it makes sense to have them together
    * Should have toggles, like `confirm-actions` has now
* Merge `scratch-messaging`, `scratch-notifier` and `msg-count-badge`:
    * They are all for messaging and don't affect Scratch pages.
    * There would be toggles for notifications and message count
* Merge `forum-search` and `search-profile` into `search`:
    * Both toggleable, of course
    * Forum search would be changed to use the existing search bar
* Merge `exact-count`, `scratchtats` and `user-id` into `stats`:
    * With toggles
    * They are all used to check stats and other data

## Merge settings
* In all addons that have them, merge costumes and sounds into assets. They're the same.

**Comment by Samq64** [comment]
> Merge `mute-project` and `vol-slider` into `vol-slider`

I agree, but WorldLanguages thought it would be better seperated. See https://github.com/ScratchAddons/ScratchAddons/pull/4734#issuecomment-1191955003

> Merge `custom-block-text`, `custom-block-shape`, `editor-theme3`, `zebra-striping`, `transparent-orphans` into `custom-block-appearance`

I disagree. That addon would have **way** to many settings.

> Merge `block-cherry-picking` and `block-duplicate` into `more-block-controls`

I agree, but I think they were seperated intentionally.

> Merge `mouse-pos`, `clones` and `debugger` into `debugger`

I'm not so sure about this one. Maybe `mouse-pos` and `clones` could be merged, but not with `debugger`.

> Merge `editor-unshare-button` and `share-through-mystuff` into `share-unshare`

The problem is one is an editor addon while the other is a website addon.

> Merge `youtube-fullscreen` and `true-youtube-links` into `youtube`

Why not?

> Merge `confirm-actions` and `remove-sprite-confirm` into `confirm-actions`

This has the same problem as the share/unshare addons

> Merge `scratch-messaging`, `scratch-notifier` and `msg-count-badge`

They could be, but not `scratch-notifier` since it requires an optional permission.

> Merge `forum-search` and `search-profile` into `search`

There are more search addons than those. I'm sur eif merging them all is a good idea though.

Merge `exact-count`, `scratchtats` and `user-id` into `stats`

I don't think these should be merged either.

> In all addons that have them, merge costumes and sounds into assets. They're the same.

I guess they could be merged

**Reply by Secret-chest** [reply]
> I disagree. That addon would have way to many settings.

If we add setting groups it won't.

**Reply by Secret-chest** [reply]
> The problem is one is an editor addon while the other is a website addon.

Scratch Addons already supports mixed addons. It would make sense.

**Reply by Secret-chest** [reply]
> I agree, but WorldLanguages thought it would be better seperated. See #4734 (comment)

Why though? The volume slider would be toggleable

**Reply by mxmou** [reply]
> > The problem is one is an editor addon while the other is a website addon.
> 
> Scratch Addons already supports mixed addons. It would make sense.

There's currently no way to make an addon appear under both "editor features" and "website features" on the settings page.

**Comment by mxmou** [comment]
Separate addons are generally easier to find - addons can be filtered using the search bar and category selector but settings can't.

> * In all addons that have them, merge costumes and sounds into assets. They're the same.

They're similar but definitely not the same. Merging these settings would just remove functionality for no reason.

**Reply by Secret-chest** [reply]
Does anyone ever set them to a different value?

**Reply by Secret-chest** [reply]
We should make settings filterable then.

**Comment by WorldLanguages** [comment]
Some of these could make sense if we add groups of settings (#5719) and setting descriptions (#5320)
Even better if every setting can have an attached screenshot.
And mxmou is right about search.
